### PR TITLE
Offline test improvements

### PR DESCRIPTION
**Pull request by palana**
These changes make the offline test deterministic in my testing, i.e. the same input file produces the same `segment_label`s in `segments.json`

### CODE REVIEW

File: src/tests/localvocal-offline-test.cpp [hunk 1]
```diff
@@ -441,31 +453,63 @@ int wmain(int argc, wchar_t *argv[])
...
+						std::lock_guard<std::mutex> lock(
+							gf->whisper_buf_mutex);
+						wait = gf->input_buffers->size != 0;
+					}
+					if (!wait)
+						break;
+
+					// sleep up to window size in case whisper is processing, so the buffer builds up similar to OBS
+					auto now = std::chrono::system_clock::now();
+					if (false && now > max_wait)
```
Comment (palana) [inline src/tests/localvocal-offline-test.cpp:470]: kind of undecided if this `false` should be a parameter, or whether the max wait should be removed

for running tests deterministically between machines you currently want to always only feed a single chunk of audio into `gf->input_buffers`

honoring `max_wait` gives you something that is closer to the "plugin within obs experience", i.e. while whisper inference (etc) is running (which takes longer than the wait time on my machine), audio buffers continue to fill up, so on the next run of the whisper loop a bigger chunk of audio is fed into VAD

File: src/tests/localvocal-offline-test.cpp [hunk 2]
```diff
@@ -43,9 +44,14 @@ void obs_log(int log_level, const char *format, ...)
 	std::time_t now_time_t = std::chrono::system_clock::to_time_t(now);
 	std::tm now_tm = *std::localtime(&now_time_t);
```
Comment (royshil) [inline src/tests/localvocal-offline-test.cpp:45]: think we don't need this anymore right?

**Review by royshil — APPROVED** [review]
looks great!

File: src/tests/localvocal-offline-test.cpp (hunk 2 above)
Comment (palana) [reply src/tests/localvocal-offline-test.cpp:45]: I kind of like having both timestamps available, local time to orient myself on when a particular run happened and "running time" to compare relative timing within a run

**Pull request merged**